staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,289 @@
+		if _, _, err := rootCmd.Find([]string{alias.Name}); err == nil {
+			fmt.Fprintf(errOut, "Warning: Setting alias %q to a built-in command is not supported\n", alias.Name)
+			continue
+		}
+
+		if _, ok := aliasArgsMap[alias.Name]; ok {
+			fmt.Fprintf(errOut, "alias %s is already set, skipping...\n", alias.Name)
+			continue
+		}
+
+		commands := strings.Split(alias.Command, " ")
+		existingCmd, flags, err := rootCmd.Find(commands)
+		if err != nil {
+			fmt.Fprintf(errOut, "Warning: command %q not found to set alias %q: %v\n", alias.Command, alias.Name, flags)
+			continue
+		}
+
+		newCmd := *existingCmd
+		newCmd.Use = alias.Name
+		aliasCmd := &newCmd
+
+		aliasArgsMap[alias.Name] = struct {
+			args    []string
+			flags   []v1alpha1.PreferencesCommandOverrideFlag
+			command *cobra.Command
+		}{
+			args:    alias.Args,
+			flags:   alias.Flags,
+			command: aliasCmd,
+		}
+	}
+
+	// It is verified that all the aliases are valid. So that, now we
+	// can define them in to the root command.
+	for key, val := range aliasArgsMap {
+		p.aliases[key] = struct{}{}
+		rootCmd.AddCommand(val.command)
+	}
+
+	aliasName := args[1]
+
+	foundAliasCmd, _, err := rootCmd.Find([]string{aliasName})
+	if err != nil {
+		return args, nil
+	}
+
+	aliasArgs, ok := aliasArgsMap[aliasName]
+	if !ok {
+		return args, nil
+	}
+
+	// This function triggers merging the persistent flags in the parent commands.
+	_ = foundAliasCmd.InheritedFlags()
+
+	for _, fl := range aliasArgs.flags {
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
+	return args, nil
+}
+
+// DefaultGetPreferences returns v1alpha1.KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
+// where kubeconfig's defaults are residing in.
+func DefaultGetPreferences(kuberc string) (*v1alpha1.Preferences, error) {
+	if !util.KubeRC.IsEnabled() {
+		return nil, nil
+	}
+
+	kubeRCFile := RecommendedKubeRCFile
+	explicitly := false
+	if kuberc != "" {
+		kubeRCFile = kuberc
+		explicitly = true
+	}
+
+	if kubeRCFile == "" && os.Getenv("KUBERC") != "" {
+		kubeRCFile = os.Getenv("KUBERC")
+		explicitly = true
+	}
+
+	kubeRCBytes, err := os.ReadFile(kubeRCFile)
+	if err != nil {
+		if os.IsNotExist(err) && !explicitly {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	decoded, err := runtime.Decode(scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion), kubeRCBytes)
+	if err != nil {
+		return nil, err
+	}
+	return decoded.(*v1alpha1.Preferences), nil
+}
+
+func getExplicitKuberc(args []string) string {
```

Comment:
Probably worth explaining why we need to manually search the flag name rather than use the built-in `Lookup` method.

Reply:
Added an explanation about why we should parse manually